src/dvr/dvr_rec.c

```diff
@@ -607,7 +611,7 @@ pvr_generate_filename(dvr_entry_t *de, const streaming_start_t *ss)
       break;
     *(dirsep - 1) = '\0';
     if (*x) {
-      s = cleanup_filename(cfg, x);
+      s = cleanup_filename(cfg, x, de->de_directory == NULL);
```

Comment:
You cannot do 'de->de_directory == NULL' check. It affects also other directory components (like title/channel subdirectories).

There is no easy fix for this issue, because at this point, you don't know which part of the path is the de_directory.